fastlane/lanes/release.rb

```diff
@@ -118,31 +221,75 @@
   #
   desc 'Creates a new hotfix branch for the given `version:x.y.z`. The branch will be cut from the `x.y` tag.'
   lane :new_hotfix_release do |options|
-    prev_ver = ios_hotfix_prechecks(options)
-    ios_bump_version_hotfix(
-      previous_version: prev_ver,
-      version: options[:version]
+    new_version = options[:version] || UI.input('Version number for the new hotfix?')
+    hotfix_build_code = hotfix_build_code(release_version: new_version)
+
+    # Parse the provided version into an AppVersion object
+    parsed_version = VERSION_FORMATTER.parse(new_version)
+    previous_version = VERSION_FORMATTER.release_version(VERSION_CALCULATOR.previous_patch_version(version: parsed_version))
+
+    # Check versions
+    message = <<-MESSAGE
+
+      New Hotfix:
+
+      • Current release version and build code: #{current_release_version} (#{current_build_code}).
+      • New release version and build code: #{new_version} (#{hotfix_build_code}).
+
+      • Current internal release version and build code: #{current_internal_release_version} (#{current_internal_build_code}).
+      • New internal release version and build code: #{new_version} (#{internal_hotfix_build_code(release_version: new_version)}).
+
+      Branching from tag: #{previous_version}
+
+    MESSAGE
+
+    if options[:skip_confirm]
+      UI.message(message)
+    else
+      UI.user_error!('Aborted by user request') unless UI.confirm("#{message}Do you want to continue?")
+    end
```

Comment:
Have you considered relying on Fastlane to manage new lines? That is, `UI.message` already know how to separate lines at the end of what it prints. We could add the trailing new line only in the `UI.confirm` call:

```suggestion
    message = <<-MESSAGE
      New Hotfix:

      • Current release version and build code: #{current_release_version} (#{current_build_code}).
      • New release version and build code: #{new_version} (#{hotfix_build_code}).

      • Current internal release version and build code: #{current_internal_release_version} (#{current_internal_build_code}).
      • New internal release version and build code: #{new_version} (#{internal_hotfix_build_code(release_version: new_version)}).

      Branching from tag: #{previous_version}
    MESSAGE

    if options[:skip_confirm]
      UI.message(message)
    else
      UI.user_error!('Aborted by user request') unless UI.confirm("#{message}\nDo you want to continue?")
    end
```

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/commit/21bf9fd0724eebb10be848aa26f5016401f4db89 and https://github.com/wordpress-mobile/WordPress-iOS/commit/6c57d7971d178a1701558c3a11fef5f79a9ad940. 

See https://github.com/wordpress-mobile/WordPress-iOS/pull/21666#discussion_r1350991187